addons/account_edi_peppol/models/account_edi_format.py

```diff
@@ -0,0 +1,220 @@
+            errors.append(_('Only companies with a European VAT are supported'))
+            # See PEPPOL_COUNTRY_EAS : only those VAT number are supported.
+            # Other identification are possible but not yet implemented in Odoo, see : https://docs.peppol.eu/poacc/billing/3.0/codelist/eas/
+        if not invoice.company_id.partner_id.commercial_partner_id.peppol_endpoint:
+            errors.append(_('Please set a vat number on the seller\'s company'))
+
+        if not invoice.commercial_partner_id.peppol_endpoint_scheme:
+            errors.append(_('Only companies with a European VAT are supported (customer)'))
+        if not invoice.commercial_partner_id.peppol_endpoint:
+            errors.append(_('Please set a vat number on the customer\'s company'))
+        return errors
+
+    def _export_peppol(self, invoice):
+        self.ensure_one()
+
+        def format_monetary(amount, currency=invoice.currency_id):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.85000000000001).
+            return float_repr(amount, currency.decimal_places)
+
+        def convert_monetary(amount, from_currency):
+            # All monetary should be in the invoice currency, except for vat total
+            return from_currency._convert(amount, invoice.currency_id, invoice.company_id, invoice.invoice_date)
+
+        def get_tax_total():
+            breakdown = {}
+            for line in invoice.invoice_line_ids.filtered(lambda line: not line.display_type):
+                if line.tax_ids:
+                    price_unit_wo_discount = line.price_unit * (1 - (line.discount / 100.0))
+                    line_taxes = line.tax_ids.compute_all(
+                        price_unit_wo_discount,
+                        quantity=line.quantity,
+                        product=line.product_id,
+                        partner=invoice.partner_id,
+                        currency=line.currency_id,
+                        is_refund=invoice.move_type in ('out_refund', 'in_refund'))['taxes']
+                    for tax in line_taxes:
+                        tax_category = 'S' if tax['amount'] else 'Z'
+                        tax_percent = self.env['account.tax'].browse(tax['id']).amount
+                        breakdown.setdefault((tax_category, tax_percent), {'base': 0, 'amount': 0})
+                        breakdown[(tax_category, tax_percent)]['base'] = tax['base']
+                        breakdown[(tax_category, tax_percent)]['amount'] = tax['amount']
+                else:
+                    breakdown.setdefault(('Z', 0), {'base': 0, 'amount': 0})
+                    breakdown[('Z', 0)]['base'] += line.price_subtotal
+
+            return {'amount': invoice.amount_tax,
+                    'seller_currency': invoice.company_id.currency_id,
+                    'amount_seller_currency': invoice.currency_id._convert(invoice.amount_tax, invoice.company_id.currency_id, invoice.company_id, invoice.date),
```

Comment:
use `invoice.amount_tax_signed` instead! (don't forget to inverse the amount in case of refund